syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -845,6 +845,31 @@ def linear(*args):
 
                 module.linear = linear
 
+                def dropout(input, p=0.5, training=True, inplace=False):
+                    """
+                    The dropout class calls functional dropout. Hence overloading functional dropout
+                    so that even works for dropout layer class.
+                    Ref:  https://stackoverflow.com/questions/54109617/implementing-dropout-from-scratch
+                    """
+                    if training:
+                        binomial = torch.distributions.binomial.Binomial(probs=1 - p)
+
+                        # we must convert the normal tensor to fixed precision before multiplication
+                        noise = (
+                            (
+                                binomial.sample(input.shape).type(torch.FloatTensor)
+                                * (1.0 / (1.0 - p))
```

Comment:
Quick question here, shouldn't you multiply by `p` instead ?
I'm no dropout expert but I just read this: https://machinelearningmastery.com/dropout-for-regularizing-deep-neural-networks/

Reply:
We are using inverted dropout which uses (1/(1-p)) to scale up the result.
Ref: https://www.coursera.org/lecture/deep-neural-network/dropout-regularization-eM33A

But would like to get it reviewed by someone :smiley:

Reply:
Excellent you're right!
https://pgaleone.eu/deep-learning/regularization/2017/01/10/anaysis-of-dropout/ confirms this